pkg/meta/model/job_args.go

```diff
@@ -1295,3 +1308,376 @@ func (a *FlashbackClusterArgs) decodeV1(job *Job) error {
 func GetFlashbackClusterArgs(job *Job) (*FlashbackClusterArgs, error) {
 	return getOrDecodeArgs[*FlashbackClusterArgs](&FlashbackClusterArgs{}, job)
 }
+
+// DropIndexArgs is the argument for drop index.
+// IndexIDs may have different length with IndexNames and IfExists.
+// When it's used in dropping index, length of IndexNames, IfExists and IndexIDs are 1.
+// When it's used in rolling back add indexes,
+// len(IndexNames) and len(IfExists) equals to the number of merged added index jobs,
+// while IndexIDs stores the partition IDs of the corresponding table.
+type DropIndexArgs struct {
+	IndexNames []pmodel.CIStr `json:"index_names,omitempty"`
+	IfExists   []bool         `json:"if_exists,omitempty"`
+
+	// Belows are used for finished args.
+	IndexIDs     []int64 `json:"index_ids,omitempty"`
+	PartitionIDs []int64 `json:"partition_ids,omitempty"`
+	// We don't support drop vector index in multi-schema change.
+	IsVector bool `json:"is_vector,omitempty"`
```

Comment:
we should make IsVector part of some index, not all indices, to make later extend easier, although we don't have this case right now

ok to leave it in this pr, we can do it in later pr, please write a todo

Reply:
DropIndexArgs` is changed to use `IndexArg`, like `AddIndexArgs`.